### ISSUE TITLE
wizard.check_grid_docker threw a TypeError

### ISSUE THREAD

**Issue by Kariten** [OpenMined/PySyft#6890]
## Description
I met the exception below when running the notebook `01-install-wizard.ipynb` at in[7]: `wizard.check_grid_docker`.
How should I solve it?

## Screenshots
An exception occured: '>' not supported between instances of 'NoneType' and 'Version'.
Please file a bug report on GitHub Issues or in Slack #support

```json
{
    "datetime": "24/10/2022 13:40:56 PDT",
    "dependencies": {
        "ansible-playbook": "/root/.hagrid/quickstart/.venv/bin/ansible-playbook",
        "docker": "/usr/bin/docker",
        "git": "/usr/bin/git"
    },
    "docker": "Client:\n Context:    default\n Debug Mode: false\n Plugins:\n  app: Docker App (Docker Inc., v0.9.1-beta3)\n  buildx: Docker Buildx (Docker Inc., v0.9.1-docker)\n  scan: Docker Scan (Docker Inc., v0.17.0)\n\nServer:\n Containers: 0\n  Running: 0\n  Paused: 0\n  Stopped: 0\n Images: 0\n Server Version: 20.10.20\n Storage Driver: overlay2\n  Backing Filesystem: extfs\n  Supports d_type: true\n  Native Overlay Diff: true\n  userxattr: false\n Logging Driver: json-file\n Cgroup Driver: cgroupfs\n Cgroup Version: 1\n Plugins:\n  Volume: local\n  Network: bridge host ipvlan macvlan null overlay\n  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog\n Swarm: inactive\n Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc\n Default Runtime: runc\n Init Binary: docker-init\n containerd version: 9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6\n runc version: v1.1.4-0-g5fd4c4d\n init version: de40ad0\n Security Options:\n  apparmor\n  seccomp\n   Profile: default\n Kernel Version: 5.15.0-52-generic\n Operating System: Ubuntu 20.04.5 LTS\n OSType: linux\n Architecture: x86_64\n CPUs: 4\n Total Memory: 7.73GiB\n Name: ubuntu\n ID: KFVX:Q4XV:33D5:SEZM:HJWI:BUFL:KTZC:YUCI:RQU6:FKBM:LAXD:5QII\n Docker Root Dir: /var/lib/docker\n Debug Mode: false\n Registry: https://index.docker.io/v1/\n Labels:\n Experimental: false\n Insecure Registries:\n  127.0.0.0/8\n Live Restore Enabled: false\n\n",
    "environment": {
        "apple_silicon": false,
        "os": "Linux",
        "os_version": "5.15.0-52-generic",
        "platform": "linux",
        "python_version": "3.8.10",
        "uname": [
            "Linux",
            "ubuntu",
            "5.15.0-52-generic",
            "#58~20.04.1-Ubuntu SMP Thu Oct 13 13:09:46 UTC 2022",
            "x86_64",
            "x86_64"
        ]
    },
    "hagrid": "0.2.108",
    "hagrid_dev": false,
    "hagrid_path": "/root/.hagrid/quickstart/.venv/lib/python3.8/site-packages",
    "hagrid_repo_sha": "88922e794f09c381f98a9281e9df1589769fe5e1",
    "python_binary": "/root/.hagrid/quickstart/.venv/bin/python"
}
Traceback (most recent call last):
File "/root/.hagrid/quickstart/.venv/lib/python3.8/site-packages/hagrid/deps.py", line 622, in check_grid_docker
return check_deps(
File "/root/.hagrid/quickstart/.venv/lib/python3.8/site-packages/hagrid/deps.py", line 580, in check_deps
dep.check()
File "/root/.hagrid/quickstart/.venv/lib/python3.8/site-packages/hagrid/deps.py", line 174, in check
if binary_info.path and binary_info.version > version.parse(
TypeError: '>' not supported between instances of 'NoneType' and 'Version'
```

## Additional Context
Output of command `docker info`:
```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.9.1-docker)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 20.10.20
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.15.0-52-generic
 Operating System: Ubuntu 20.04.5 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 7.73GiB
 Name: ubuntu
 ID: KFVX:Q4XV:33D5:SEZM:HJWI:BUFL:KTZC:YUCI:RQU6:FKBM:LAXD:5QII
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by Kariten** [comment]
I solved this problem by manually install docker compose from https://github.com/docker/compose

**Comment by iamtrask** [comment]
@Kariten - it would be amazing if Hagrid automatically detected the wrong version of docker/docker compose and prompted you to manually install. Any chance that's something you might be interested in helping to add to the interface?

**Comment by Kariten** [comment]
> @Kariten - it would be amazing if the wizard automatically detected the wrong version of docker/docker compose and prompted you to manually install. Any chance that's something you might be interested in helping to add to the interface?

I've created a pull request which fixed this problem, you can check out #6901.
